core/validation/checks.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABC, abstractmethod
+
+import pandas as pd
+
+
+class Check(ABC):
+    """
+    Abstract base class representing a validation check on a specific tab/cell of a workbook.
+
+    Attributes:
+        sheet (str): The name of the sheet in the workbook.
+        column (int): The column index of the cell to check.
+        row (int): The row index of the cell to check.
+        expected_values (tuple): A tuple of expected values for the cell.
+        error_message (str): The error message sent to the user if the check fails.
+
+    Methods:
+        get_actual_value(workbook: dict[str, pd.DataFrame]) -> str:
+            Retrieve the actual value from the workbook.
+        run(workbook: dict[str, pd.DataFrame]) -> bool:
+            Execute the check on the provided workbook.
+    """
+
+    def __init__(self, sheet: str, column: int, row: int, expected_values: tuple, error_message: str):
+        self.sheet = sheet
+        self.column = column
+        self.row = row
+        self.expected_values = expected_values
+        self.error_message = error_message
+
+    def get_actual_value(self, workbook: dict[str, pd.DataFrame]) -> str:
+        return str(workbook[self.sheet].iloc[self.column][self.row]).strip()
+
+    @abstractmethod
+    def run(self, workbook: dict[str, pd.DataFrame], **kwargs) -> tuple[bool, str]:
+        pass
+
+
+class BasicCheck(Check):
+    """
+    Used for checks where the expected values are predefined.
+    """
+
+    def run(self, workbook: dict[str, pd.DataFrame], **kwargs) -> tuple[bool, str]:
+        result = self.get_actual_value(workbook) in self.expected_values
+        return result, self.error_message
+
+
+class DynamicCheck(Check, ABC):
+    """
+    Abstract class to represent a dynamic check, i.e. where parameters must be dynamically generated at runtime.
+    Contains an abstract method to retrieve the expected values for the check.
+    """
+
+    @abstractmethod
+    def get_expected_values(self, workbook: dict[str, pd.DataFrame], **kwargs) -> tuple:
+        pass
+
+
+class ConflictingCheck(DynamicCheck):
+    """
+    A dynamic check that checks whether the values in two cells conflict with each other, for example where the provided
+    fund type does not match any of the expected fund types for the provided place name. It uses a mapping defined
+    elsewhere in the source code to map the values from one cell to a list of expected values for the other cell.
+
+    Attributes:
+        Inherits attributes from the DynamicCheck class.
+        mapping (dict): A dictionary mapping the values from the mapped column and row to the expected values.
+        mapped_column (int): The column index of the cell to map from.
+        mapped_row (int): The row index of the cell to map from.
+    """
+
+    mapping: dict
```

Comment:
the return type and implementation `get_expected_values` suggests that we know the type of this will be `dict[str, tuple]`